lib/improver/calculate_sleet_prob.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017-2019 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""A plugin to calculate probability of sleet"""
+
+import numpy as np
+from improver.metadata.utilities import create_new_diagnostic_cube
+
+
+def calculate_sleet_probability(prob_of_snow,
+                                prob_of_rain):
+    """
+    This calculates the probability of sleet using the calculation:
+    prob(sleet) = 1 - (prob(falling_snow_level at or below surface > 0.8) +
+                       prob(falling_rain_level at or above surface > 0.8))
+
+    Args:
+      prob_of_snow_falling_level (iris.cube.Cube):
+        Cube of snow falling level probabilities
+
+      prob_of_rain_falling_level (iris.cube.Cube):
+        Cube of rain falling level probabilities
+
+    Returns:
+      iris.cube.Cube:
+        Cube of the probability of sleet
+
+        Raises:
+            ValueError: If the cube contains.
+    """
+    ones = np.ones((prob_of_snow.shape), dtype="float32")
+    sleet_prob = (ones - (prob_of_snow.data + prob_of_rain.data))
+    if np.any(sleet_prob < 0.0):
+        msg = ("Error - Negative values found in cube")
```

Comment:
I've been trying to think through whether getting > 0.5 probs for both rain and snow at surface is impossible, and I don't think it is. I am also not sure that forcing the values to 0 would be wise, as a messy rapid thawing layer is quite likely to produce some sleet.

Errors are likely to catch us out and make things fall over, but I don't know that we would notice warnings. As such I think an error is probably right until we start to test this out, at which point we might need to decide on an alternative course of action.

Could you please make the error more descriptive, e.g. ```Probabilities of sleet have been calculated as negative values.``` or some such.